aptos-move/framework/src/module_metadata.rs

```diff
@@ -241,7 +244,9 @@ pub fn check_metadata_format(module: &CompiledModule) -> Result<(), MalformedErr
                 bcs::from_bytes::<RuntimeModuleMetadataV1>(&data.value)
                     .map_err(|e| MalformedError::DeserializedError(data.key.clone(), e))?;
             }
-        } else if data.key == *COMPILATION_METADATA_KEY {
+        } else if features.is_enabled(FeatureFlag::REJECT_UNSTABLE_BYTECODE)
```

Comment:
if the feature flag `REJECT_UNSTABLE_BYTECODE` is not enabled, unknown key will be raised as long as the compilation metadata is attached.